traffic_ops/app/lib/API/Tenant.pm

```diff
@@ -44,14 +44,16 @@ sub index {
 	my $orderby = $self->param('orderby') || "name";
 	my $rs_data = $self->db->resultset("Tenant")->search( undef, {order_by => 'me.' . $orderby } );
 	while ( my $row = $rs_data->next ) {
-		push(
-			@data, {
-				"id"           => $row->id,
-				"name"         => $row->name,
-				"active"       => \$row->active,
-				"parentId"     => $row->parent_id,
-			}
-		);
+		if (verify_tenancy_for_read($self, $row->id)) {
```

Comment:
this might cause problems. for example, if there are 200 tenants, i think this will run 400 queries. since it looks like verify_tenancy() makes 2 queries for each time it is called. i would recommend making your initial query smarter.